kornia/geometry/liegroup/se3.py

```diff
@@ -247,3 +247,22 @@ def inverse(self) -> 'Se3':
         """
         r_inv = self.r.inverse()
         return Se3(r_inv, r_inv * (-1 * self.t))
+
+    def adjoint(self) -> Tensor:
+        """Returns the adjoint matrix of shape :math:`(B, 6, 6)`.
+
+        Example:
+            >>> s = Se3.identity()
+            >>> s.adjoint()
+            tensor([[[1., 0., 0., 0., 0., 0.],
+                     [0., 1., 0., 0., 0., 0.],
+                     [0., 0., 1., 0., 0., 0.],
+                     [0., 0., 0., 1., 0., 0.],
+                     [0., 0., 0., 0., 1., 0.],
+                     [0., 0., 0., 0., 0., 1.]]], grad_fn=<CatBackward0>)
```

Comment:
why it returns batched while in theory you are calling unbatched identity ?

Reply:
It is internally represented as batched

Reply:
then not sure it's correct. If Se3 data storage is non-batched, all the chained ops should keep the same data layout

Reply:
Should it return as unbatched for unbatched inputs? Internally representing it as batched reduces a bit of checks, this why i did this way.

Reply:
> then not sure it's correct. If Se3 data storage is non-batched, all the chained ops should keep the same data layout

okay